sequence.go

```diff
@@ -0,0 +1,24 @@
+package dbus
+
+// Sequence represents the value of a monotonically increasing counter.
+type Sequence uint64
+
+const (
+	// NoSequence indicates the absence of a sequence value.
+	NoSequence Sequence = 0
+)
+
+// sequenceGenerator represents a monotonically increasing counter.
+type sequenceGenerator struct {
+	nextSequence Sequence
+}
+
+func (generator *sequenceGenerator) next() Sequence {
+	result := generator.nextSequence
+	generator.nextSequence++
```

Comment:
Does roll over matter?

Reply:
Yes. In the code to implement race-free get initial state+subscribe, we compare the sequence number of the response advising us the initial state of the object with the sequence number(s) of event(s) advising us of a stage change. We discard events with a sequence number prior to the response containing the initial state. If we rollover, these comparisons could be invalid. 
I chose Sequence to be backed by a uint64 as we are unlikely to be able to overflow such a number in sequential execution. Even at 10 billion increments per second, it would take ~58 years to overflow.